trellis/containers/transforms.hpp

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (C) 2022 Agtonomy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef TRELLIS_CONTAINERS_TRANSFORMS_HPP
+#define TRELLIS_CONTAINERS_TRANSFORMS_HPP
+
+#include <Eigen/Geometry>
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+#include "trellis/core/time.hpp"
+
+namespace trellis {
+namespace containers {
+
+/**
+ * Transforms a container to hold rigid coordinate frame transformations
+ *
+ * This class holds transforms with associated timestamps. It also maintains a validity window, which is a duration of
+ * time that the transform is valid for. Lookups are done based on a timestamp. The algorithm will first find the
```

Comment:
Worth noting that the validity window applies both before and after a transform's timestamp?